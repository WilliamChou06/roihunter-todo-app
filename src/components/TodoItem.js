import React, { Component } from 'react';
import { connect } from 'react-redux';
import { toggleTodo } from '../actions';
import { List, Checkbox } from 'antd';
import styled from 'styled-components';

const Todo = styled(List.Item)`
  cursor: pointer;
  transition: all 0.3s;
  padding-left: 5px;

  &:hover {
    background: #f1f1f1;
  }
`;

class TodoItem extends Component {

  handleToggleTodo = (id) => {
    this.props.toggleTodo(id)
  }

  render() {
    return (
      <Todo onClick={() => {
        this.handleToggleTodo(this.props.todoId)
      }}>
        {this.props.children}
        <Checkbox type="danger" checked={this.props.completed} />
      </Todo>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  toggleTodo: id => dispatch(toggleTodo(id))
});



export default connect(undefined, mapDispatchToProps)(TodoItem);
