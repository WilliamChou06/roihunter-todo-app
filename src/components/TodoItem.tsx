import React, { Component } from 'react';
import { connect } from 'react-redux';
import { startToggleTodo } from '../actions';
import { List, Checkbox } from 'antd';
import styled from 'styled-components';

const Todo = styled(List.Item)`
  cursor: pointer;
  transition: all 0.3s;
  padding-left: 5px;
  text-decoration-line: ${props => (props.isCompleted ? 'line-through' : 'none')};
  display: flex;
  justify-content: space-between;
`;

const StyledSpan = styled.span`
  padding-right: 6px;
`

interface Props {
  todoId: string;
  isCompleted: boolean;
  children: any;
  startToggleTodo(id: string): any;
}
class TodoItem extends Component<Props> {
  handleToggleTodo = id => {
    this.props.startToggleTodo(id);
  };

  render() {
    const { todoId, isCompleted, children } = this.props;
    return (
      <Todo
        onClick={() => {
          this.handleToggleTodo(todoId);
        }}
        isCompleted={isCompleted}
      >
        <StyledSpan>{children}</StyledSpan>
        <Checkbox type="danger" checked={isCompleted} />
      </Todo>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  startToggleTodo: id => dispatch(startToggleTodo(id))
});

export default connect(
  undefined,
  mapDispatchToProps
)(TodoItem);
