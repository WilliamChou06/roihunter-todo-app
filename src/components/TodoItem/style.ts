import styled from 'styled-components';
import { List } from 'antd';

export const Todo = styled(List.Item)`
  cursor: pointer;
  transition: all 0.3s;
  padding-left: 5px;
  text-decoration-line: ${props =>
    props.isCompleted ? 'line-through' : 'none'};
  display: flex;
  justify-content: space-between;
`;

export const StyledSpan = styled.span`
  padding-right: 6px;
`;
