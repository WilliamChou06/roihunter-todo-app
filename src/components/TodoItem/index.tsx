import React from 'react';
import { useDispatch } from 'react-redux';
import { startToggleTodo } from '../../actions';
import { Checkbox } from 'antd';
import { Todo, StyledSpan } from './style';

interface Props {
  todoId: string;
  isCompleted?: boolean;
  children: any;
}
const TodoItem: React.FC<Props> = ({ todoId, isCompleted, children }) => {
  const dispatch = useDispatch();

  const handleToggleTodo = id => {
    dispatch(startToggleTodo(id));
  };

  return (
    <Todo
      onClick={() => {
        handleToggleTodo(todoId);
      }}
      isCompleted={isCompleted}
    >
      <StyledSpan>{children}</StyledSpan>
      <Checkbox checked={isCompleted} />
    </Todo>
  );
};

export default TodoItem;
