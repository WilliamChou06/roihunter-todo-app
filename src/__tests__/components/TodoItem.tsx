import React from 'react';
import { shallow } from 'enzyme';
import TodoItem from '../../components/TodoItem';

describe('snapshot test', () => {
  it('should render TodoItem component correctly', () => {
    // @ts-ignore
    const wrapper = shallow(<TodoItem />);
    expect(wrapper).toMatchSnapshot();
  });
});
