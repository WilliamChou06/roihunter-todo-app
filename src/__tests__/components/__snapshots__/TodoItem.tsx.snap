// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test should render TodoItem component correctly 1`] = `ShallowWrapper {}`;
